### PR TITLE
Move authentication tracking to SDK facade

### PR DESCRIPTION
**Pull request by marcospassos**
## Summary
Move authentication tracking to SDK facade.

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/facade/sdkFacade.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export default class SdkFacade {
         if (userId !== undefined) {
             sdk.identify(userId);
         } else if (token !== undefined) {
-            sdk.setToken(token);
+            sdk.setToken(Token.parse(token));
```
[the remark this answers is not in the document]
Comment (marcospassos) [reply src/facade/sdkFacade.ts:75]: Me neither. How about a warning?

File: test/facade/sdkFacade.test.ts [hunk 2]
```diff
@@ -340,56 +346,180 @@ describe('A SDK facade', () => {
...
+        const carolToken = Token.issue(appId, 'c4r0l');
+
+        expect(sdkFacade.getToken()).toBeNull();
+
+        sdkFacade.setToken(carolToken);
+
+        expect(sdkFacade.getToken()).toEqual(carolToken);
+    });
+
+    test('should allow to refresh the token of the current user', () => {
```
Comment (marcospassos) [inline test/facade/sdkFacade.test.ts:431]: Here

Comment (Fryuni) [reply test/facade/sdkFacade.test.ts:431]: This tests refreshing a identified token, sending only one `userSignedIn`. We are missing a test refreshing anonymous token, that should not send any tracked event.

Comment (marcospassos) [reply test/facade/sdkFacade.test.ts:431]: oh, got it

File: src/facade/sdkFacade.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcospassos) [reply src/facade/sdkFacade.ts:75]: ...besides switching the order

[the remark this answers is not in the document]
Comment (Fryuni) [reply src/facade/sdkFacade.ts:75]: I would be ok with a warning. But with token taking precedence over plain userId

[the remark this answers is not in the document]
Comment (marcospassos) [reply src/facade/sdkFacade.ts:75]: > The user id was ignored in favor of the specified token. Make sure to specify either the "userId" or "token" option, but not both.

[the remark this answers is not in the document]
Comment (Fryuni) [reply src/facade/sdkFacade.ts:75]: I agree.

**Pull request merged**